File: lab/javascript/somas/lambda_soma.js
import assert from "node:assert";  // ou const assert = require("node:assert");

// forma curta (expression body)
const soma = (x, y) => x + y;

const valores_e_saidas = [
    [1, 1, 2],
    [4, 5, 9],
    [10, 5, 15],
    [-1, 1, 0],
    [-2, -2, -4],
    [3, 7, 10]
];

for (const valor_e_saida of valores_e_saidas){
    const [x, y, saida] = valor_e_saida;
    const saida_soma = soma(x, y);
    console.log(saida_soma);
    const erro_saida = `SOMA INVÁLIDA: ${x} + ${y} = ${x+y} sua saida foi: ${saida}`;
    assert.strictEqual(saida, saida_soma, erro_saida);
}

File: lab/python/somas/lambda_soma.py
# Evite usar lambdas assim, isto é apenas para estudos
soma = lambda x, y: x + y  # noqa: E731


valores_e_saidas = (
    [1, 1, 2],
    [4, 5, 9],
    [10, 5, 15],
    [-1, 1, 0],
    [-2, -2, -4],
    [3, 7, 10]
)

for valor_e_saida in valores_e_saidas:
    x, y, saida = valor_e_saida
    saida_soma = soma(x, y)
    print(saida_soma)
    erro_saida = f"SOMA INVÁLIDA: {x} + {y} = {x+y} sua saida foi: {saida}"
    assert saida == saida_soma, erro_saida
